### PR TITLE
Add TODO for Updating API Spec

### LINKED ISSUE #13619: [Feature Request] Add TODO for Updating API Spec
### Is your feature request related to a problem? Please describe

With the introduction of OpenSearch API Specification for generating client libraries and documentation in multiple languages
(See: https://github.com/opensearch-project/opensearch-api-specification?tab=readme-ov-file#welcome), it is possible that new features might miss out on defining the APIs, leading in gaps in what the generated clients for exotic languages can support.

### Describe the solution you'd like

It would be good to have  the pull request template include an item in its checklist for updating the API specs as a precondition to committing the feature.

### Related component

Clients

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kkhatua**
### Description
Added a TODO in the checklist for contributors to add API Specs when introducing new user APIs

### Related Issues
Resolves #13619

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ Resolves #[Issue number to be closed when this PR is merged]
   - [ ] All tests pass
 - [ ] New functionality has been documented.
   - [ ] New functionality has javadoc added
+- [ ] Specifications for new APIs to support generated clients have been added (See: [Guide](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md))
```
Comment (peternied) [inline .github/pull_request_template.md:20]: Lets we make this a single checkbox item, similar to the docs issue at the bottom of this checklist.  What do you think?

```suggestion
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
```

Comment (kkhatua) [reply .github/pull_request_template.md:20]: I have no strong opinions on this. The only thing is that we need to make sure if there is a need to update the API spec, there should be a way for the `API Spec` maintainers to catching this, like the doc team does.

**Review by peternied — APPROVED** [review]
Looks good to me @reta @andrross any concerns?

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
DCO check is failing :(

**Comment by peternied** [comment]
I've approved and squashed it with correct sign-off statements

**Comment by kkhatua** [comment]
Thanks @reta  and @peternied ! 
I didn't realize that non-code changes to the checklist also required the entire checklist. #ironic 😇 

Will take care and follow protocol

**Pull request merged**